node/src/chain.rs

```diff
@@ -528,20 +366,272 @@ pub async fn create_ethereum_networks_for_chain(
             ),
             web3.limit_for(&config.node),
         );
+
+        if call_only {
+            call_only_adapters.push(adapter);
+        } else {
+            adapters.push(adapter);
+        }
     }
 
-    parsed_networks.sort();
-    Ok(parsed_networks)
+    adapters.sort_by(|a, b| {
+        a.capabilities
+            .partial_cmp(&b.capabilities)
+            // We can't define a total ordering over node capabilities,
+            // so incomparable items are considered equal and end up
+            // near each other.
+            .unwrap_or(Ordering::Equal)
+    });
```

Comment:
This is the second time this makes an appearance. Would be better to move that to some helper method. And add a comment what this ordering is and why it is important.